index.js

```diff
@@ -2,7 +2,19 @@ const assert = require('assert')
 
 module.exports = cacheElement
 
-// Memoize a bel element
-// null -> null
-function cacheElement () {
+function cacheElement (fn) {
+  const store = {}
+  
+  return function render () {
+    const args = Array.from(arguments)
+    const argsAreTheSame = JSON.stringify(store.prev) === JSON.stringify(args)
+    
+    if (argsAreTheSame) return store.el
```

Comment:
In order for this to work we can't return the `el` directly as a Node can't exist in two different DOM trees at the same time. Instead we need to return a proxy element [as per this repo](https://github.com/yoshuawuyts/playground-stateful-choo/blob/master/index.js#L70-L80) - hope this makes sense hah